### PR TITLE
Make template replacer write even if no target field exists

### PR DESCRIPTION
**Pull request by ppcad**
Make template replacer work even if target field does not exist

Before, the template replacer only wrote into an already existing target field.
This can be a problem if it is desired to drop that field before the template replacer processes the event.
Therefore, the template replacer now writes the target field even if it was dropped before.
Only replacing the target field it can be still achieved by filtering for the existence of that field.

### CODE REVIEW

File: logprep/processor/template_replacer/processor.py [hunk 1]
```diff
@@ -140,11 +140,21 @@ def _apply_rules(self, event, rule):
...
+                else:
+                    raise TemplateReplacerError(
+                        self.name,
+                        f"Parent field '{subfield}' of target field '{self._target_field}' "
+                        f"exists and is not a dict!",
+                    )
+            _event[self._target_field_split[-1]] = _dict
 
     @staticmethod
     def _field_exists(event: dict, dotted_field: str) -> bool:
```
[the remark this answers is not in the document]
Comment (ppcad) [reply logprep/processor/template_replacer/processor.py]: I removed it.

File: logprep/processor/template_replacer/processor.py [hunk 2]
```diff
@@ -140,11 +140,21 @@ def _apply_rules(self, event, rule):
...
+                    _event[subfield] = {}
                     _event = _event[subfield]
-                _event[self._target_field_split[-1]] = _dict
+                else:
+                    raise TemplateReplacerError(
+                        self.name,
+                        f"Parent field '{subfield}' of target field '{self._target_field}' "
+                        f"exists and is not a dict!",
+                    )
+            _event[self._target_field_split[-1]] = _dict
```
[the remark this answers is not in the document]
Comment (ppcad) [reply logprep/processor/template_replacer/processor.py]: I changed it. I usually kept longer methods to improve the performance. All those calls etc. start to add up.
We will see how the performance of logprep changes in the long run.

[the remark this answers is not in the document]
Comment (ekneg54) [reply logprep/processor/template_replacer/processor.py]: I think splitting up functions will increase the performance of the developer who has to read it. ;-)

**Review by dtrai2 — APPROVED** [review]
Thanks for the changes :)

**Pull request merged**